File: tourplanner/src/app/dto/TourDto.ts
import { Log } from "../models/log.model";
import { TourRoute } from "../models/tourRoute.model";
import { TourType } from "../models/tour.model";

export type TourDto = {
    id: number;
    name: string;
    description: string;
    estimated_time: number;
    popularity: number;
    isChildfriendly: boolean;
    tourType: string;
    routes: TourRoute[];
    logs: Array<Log & { createdAt: string }>;
}

File: tourplanner-backend/src/main/java/org/tour/tourplannerbackend/dto/TourDto.java
package org.tour.tourplannerbackend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.tour.tourplannerbackend.model.enums.TourType;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TourDto {
    private Long id;
    private String name;
    private String description;
    private TourType tourType;
}

